### ISSUE TITLE
Implement a timeout warning 

### ISSUE THREAD

**Issue by quarterback** [18F/cms-hitech-apd#1666]
Timeout happens after 1 hour of signed in activity. If so, consider:

- Letting users know there is an inactivity timeout
- Giving them a warning/chance to save before logging out
- Extending the time limit or allowing it to be adjusted.

**Comment by mgwalker** [comment]
This is particularly important for users that may have switched to another tab for a little while, so the notification can draw their attention back if they want to stay logged in. 

We probably need to find a way to synchronize across tabs in case a user has reopened the app in a second tab - we don't want their first tab to log them out while they're actively using the second tab.

**Comment by mgwalker** [comment]
Here's my thinking at the moment:

- we can create a [BroadcastChannel](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) so that multiple tabs with eAPD open in them can communicate, so that activity in any eAPD tab is enough to reset the idle logout timer in all of the tabs
   > Does not work in IE, Edge, or Safari
- we can use the [Notification API](https://developer.mozilla.org/en-US/docs/Web/API/notification) to pop up a notification when the user will soon be logged out
   > Does not work in IE, tricky to make it work in other browsers
- we can flash the tab title to get the user's attention
   ![flashing tab title](https://user-images.githubusercontent.com/1775733/61150997-6bb60f80-a4aa-11e9-995e-cd84b6017754.gif)
   > Works in all the browsers
